src/main/java/com/oath/halodb/HaloDBInternal.java

```diff
@@ -410,6 +414,15 @@ long size() {
         return inMemoryIndex.size();
     }
 
+    int size(byte[] key) {
+        InMemoryIndexMetaData metaData = inMemoryIndex.get(key);
+        if (metaData == null) {
+            return 0;
```

Comment:
perhaps this should be -1 since 0 is a valid value size.  Then -1 indicates the key is not present, and 0 indicates a value of size 0.

Reply:
Good point, fixed. Method `HaloDB.get` also returns `0` for keys that are not stored. I was following the same logic. But I agree that it should return `-1` for abscent keys.

Reply:
Interesting... HaloDB.get returns a byte[], but HaloDBInternal has another get method that takes a ByteBuffer and reads the value into it (or truncating it if the value is larger than the buffer).   This returns the number of bytes copied into the buffer, and 0 when not present.

However, this method is not used anywhere.  Its not tested, its not used by the public facing API, it seems to be dead (and untested!) code.